### PR TITLE
MQTT Blinds Initial PR

### PR DESCRIPTION
**Pull request by jmichaelcopino**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,7 +43,7 @@ mqtt:
   discovery: true
   discovery_prefix: homeassistant
   topic_prefix: mqtt-blinds
-  availability_topic: mqtt-blinds/available
+  availabilityTopic: mqtt-blinds/available
```
Comment (matthew-larner) [inline README.md]: please change back to snake_case

File: config/mosquitto.conf [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#pid_file /var/run/mosquitto.pid
+
+persistence true
+persistence_location /mosquitto/data/
+
+log_dest file /mosquitto/log/mosquitto.log
+log_dest stdout
+
+#password_file /mosquitto/config/mosquitto.passwd
+allow_anonymous true
```
Comment (matthew-larner) [inline config/mosquitto.conf]: Please remove mosquito from this repo. Functional tests aren't required (only unit tests are required and shouldn't connect to MQTT).

File: src/entities/homeAssistantHandler.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+              payload = {
+                name: `${name}`,
+                unique_id: blindName,
+                command_topic: `${mqttConfig.topic_prefix}/cover/${blindName}/config`,
+                position_topic: `${mqttConfig.topic_prefix}/${blindName}/position`,
+                set_position_topic: `${mqttConfig.topic_prefix}/${blindName}/position/set`,
+                availability_topic: `${mqttConfig.availability_topic}`,
+                device_class: type == "blind" ? type : "awning",
+                payload_stop: type == "blind" ? null : "stop",
```
Comment (matthew-larner) [inline src/entities/homeAssistantHandler.ts:44]: missing some variables. Please double check the lucid chart doc

File: src/entities/rollerBlindsHandler.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+export const rollerBlindsCommandsHandler =
+  ({ mqttClient, hubs, mqttConfig }: RollerBlindHandler) =>
+  async (data: Buffer) => {
+    const message = data.toString().replace(/\s/g, "");
+
+    const bridge_address = message.substring(1, 4);
+    const motor_address = message.substring(4, 7);
+    const action = message.substring(7, 8);
+    const position = message.substring(8, 11);
```
Comment (matthew-larner) [inline src/entities/rollerBlindsHandler.ts]: This won't work. The bridge_address/motor_address won't always be 3 characters.

**Review by matthew-larner — CHANGES_REQUESTED** [review]

File: src/entities/homeAssistantHandler.ts (hunk 3 above)
Comment (jmichaelcopino) [reply src/entities/homeAssistantHandler.ts:44]: this is done

File: src/entities/rollerBlindsHandler.ts (hunk 4 above)
Comment (jmichaelcopino) [reply src/entities/rollerBlindsHandler.ts]: as for solution, I add  "|" as a separator.  then will be remove later on during mqtt publish.

File: config/mosquitto.conf (hunk 2 above)
Comment (jmichaelcopino) [reply config/mosquitto.conf]: this is untracked

**Pull request closed without merging**